aws-crt-kotlin/native/src/aws/sdk/kotlin/crt/Logging.kt

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.sdk.kotlin.crt
+
+import kotlinx.atomicfu.atomic
+import kotlinx.cinterop.*
+import libcrt.*
+
+@OptIn(ExperimentalForeignApi::class)
+internal object Logging {
+    private val initialized = atomic(false)
+
+    internal fun initialize(config: Config) {
+        if (!initialized.compareAndSet(false, true)) return
+
+        memScoped {
+            val options = cValue<aws_logger_standard_options> {
+                when (config.logDestination) {
+                    LogDestination.None -> return
+                    LogDestination.Stdout -> file = platform.posix.stdout
+                    LogDestination.Stderr -> file = platform.posix.stderr
+                    LogDestination.File -> { filename = config.logFile?.cstr?.ptr
+                        ?: throw IllegalArgumentException("LogDestination.File configured without logFile")
+                    }
+                }
+                level = config.logLevel.value.convert()
+            }
+
+            if (CrtDebug.traceLevel > 0) {
+                aws_logger_init_noalloc(s_crt_kotlin_logger.ptr, Allocator.Default.allocator, options).awsAssertOpSuccess {
+                    "failed to initialize no-alloc logger"
+                }
+            } else {
+                aws_logger_init_standard(s_crt_kotlin_logger.ptr, Allocator.Default.allocator, options).awsAssertOpSuccess {
+                    "failed to initialize standard logger"
+                }
+            }
```

Comment:
**Question**: Changing the logger implementation based on trace level seems risky since certain behaviors (and therefore _bugs_) may only appear when tracing is enabled/disabled. Is there a reason not to use the `noalloc` logger when the trace level is 0?

Reply:
This is in line with [crt-java](https://github.com/awslabs/aws-crt-java/blob/fe6356c57475685d7c0f3dae4c9d2a837eed96a9/src/native/logging.c#L61). If there are bugs then we'd open an issue against CRT no?